resilience4j-spring/src/main/java/io/github/resilience4j/bulkhead/configure/BulkheadAspect.java

```diff
@@ -85,7 +102,7 @@ private Bulkhead getBackendMonitoredAnnotation(ProceedingJoinPoint proceedingJoi
 		return AnnotationExtractor.extract(proceedingJoinPoint.getTarget().getClass(), Bulkhead.class);
 	}
 
-	private Object handleJoinPoint(ProceedingJoinPoint proceedingJoinPoint, io.github.resilience4j.bulkhead.Bulkhead  bulkhead, String methodName) throws Throwable {
+	private Object handleJoinPoint(ProceedingJoinPoint proceedingJoinPoint, io.github.resilience4j.bulkhead.Bulkhead bulkhead, String methodName) throws Throwable {
 		BulkheadUtils.isCallPermitted(bulkhead);
```

Comment:
Can we add a bulkhead.executeCheckedSupplier

    default <T> T executeCheckedSupplier(CheckedFunction0<T> checkedSupplier) throws Throwable {
        return decorateCheckedSupplier(this, checkedSupplier).apply();
    }
and use bulkhead.executeCheckedSupplier(proceedingJoinPoint::proceed)?

Reply:
Good idea !